PHP Formatting Not Working

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I've asked for help in other PHP related issues and code formatting issues I'm creating this specific because the language_server works with most files but it's not working with PHP since ZED moved PHP to an extension. But only the format on save is not working.... I had to move back to VSCODE because it stopped working and I have no idea why. Anybody?

Opening Zed, saving a PHP file... I get this in my LOG

2024-04-23T18:15:50+00:00 [INFO] ========== starting zed ==========
2024-04-23T18:15:50+00:00 [INFO] Opening main db
2024-04-23T18:15:50+00:00 [INFO] Using git binary path: Some("/Applications/Zed.app/Contents/MacOS/git")
2024-04-23T18:15:50+00:00 [INFO] extensions updated. loading 3, reloading 0, unloading 0
2024-04-23T18:15:50+00:00 [ERROR] crates/zed/src/zed.rs:628: EOF while parsing a value at line 1 column 0
2024-04-23T18:15:50+00:00 [INFO] Opening main db
2024-04-23T18:15:51+00:00 [INFO] set status on client 0: Authenticating
2024-04-23T18:15:51+00:00 [INFO] set status on client 98447: Connecting
2024-04-23T18:15:51+00:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2024-04-23T18:15:51+00:00 [INFO] Opening main db
2024-04-23T18:15:51+00:00 [INFO] Opening main db
2024-04-23T18:15:51+00:00 [INFO] set environment variables from shell:/bin/zsh, path:/Users/xcrap/.bun/bin:/opt/homebrew/opt/curl/bin:/opt/homebrew/opt/curl/bin:/opt/homebrew/opt/curl/bin:/Users/xcrap/Library/Application Support/Herd/bin/:/Users/xcrap/miniconda3/bin:/Users/xcrap/miniconda3/condabin:/opt/homebrew/bin:/opt/homebrew/sbin:/Users/xcrap/.nvm/versions/node/v20.10.0/bin:/Applications/MAMP/bin/php/php8.2.4/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin://Applications/Topaz Photo AI.app/Contents/Resources/bin:/Users/xcrap/Library/Application Support/JetBrains/Toolbox/scripts
2024-04-23T18:15:51+00:00 [INFO] starting language server "tailwindcss-language-server", path: "/Users/xcrap/htdocs/hotelgaivota", id: 1
2024-04-23T18:15:51+00:00 [INFO] starting language server "intelephense", path: "/Users/xcrap/htdocs/hotelgaivota", id: 2
2024-04-23T18:15:51+00:00 [INFO] fetching latest version of language server "tailwindcss-language-server"
2024-04-23T18:15:51+00:00 [INFO] Node runtime install_if_needed
2024-04-23T18:15:51+00:00 [INFO] Node runtime install_if_needed
2024-04-23T18:15:51+00:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-04-23T18:15:51+00:00 [INFO] Node runtime install_if_needed
2024-04-23T18:15:52+00:00 [INFO] starting language server. binary path: "/Users/xcrap/Library/Application Support/Zed/node/node-v18.15.0-darwin-arm64/bin/node", working directory: "/", args: ["/Users/xcrap/Library/Application Support/Zed/copilot/copilot-v0.5.0/dist/agent.js", "--stdio"]
2024-04-23T18:15:52+00:00 [INFO] add connection to peer
2024-04-23T18:15:52+00:00 [INFO] waiting for server hello
2024-04-23T18:15:52+00:00 [INFO] got server hello
2024-04-23T18:15:52+00:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 392, id: 4189661 })
2024-04-23T18:15:52+00:00 [INFO] set status on client 98447: Connected { peer_id: PeerId { owner_id: 392, id: 4189661 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-04-23T18:15:52+00:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-04-23T18:15:52.337Z] Agent service starting",
  "metadataStr": "[DEBUG] [agent] [2024-04-23T18:15:52.337Z]",
  "extra": [
    "Agent service starting"
  ]
}
2024-04-23T18:15:52+00:00 [INFO] Language server with id 0 sent unhandled notification client/registerCapability:
{
  "registrations": [
    {
      "id": "e963ed01-14bc-4e67-8e49-e8e9395530d4",
      "method": "workspace/didChangeWorkspaceFolders",
      "registerOptions": {}
    }
  ]
}
2024-04-23T18:15:52+00:00 [INFO] Language server with id 0 sent unhandled notification LogMessage:
{
  "level": 0,
  "message": "[DEBUG] [agent] [2024-04-23T18:15:52.867Z] Telemetry initialized",
  "metadataStr": "[DEBUG] [agent] [2024-04-23T18:15:52.867Z]",
  "extra": [
    "Telemetry initialized"
  ]
}
2024-04-23T18:15:52+00:00 [INFO] Node runtime install_if_needed
2024-04-23T18:15:53+00:00 [INFO] starting language server. binary path: "/Users/xcrap/Library/Application Support/Zed/node/node-v18.15.0-darwin-arm64/bin/node", working directory: "/Users/xcrap/htdocs/hotelgaivota", args: ["/Users/xcrap/Library/Application Support/Zed/extensions/work/php/node_modules/intelephense/lib/intelephense.js", "--stdio"]
2024-04-23T18:15:53+00:00 [INFO] Language server with id 2 sent unhandled notification window/logMessage:
{
  "type": 3,
  "message": "Initialising intelephense 1.10.4"
}
2024-04-23T18:15:53+00:00 [INFO] Language server with id 2 sent unhandled notification window/logMessage:
{
  "type": 3,
  "message": "Reading state from /var/folders/0j/629mw8tj5y7dkm056_1pthq40000gn/T/intelephense/44936fb5."
}
2024-04-23T18:15:53+00:00 [INFO] Language server with id 2 sent unhandled notification window/logMessage:
{
  "type": 3,
  "message": "Initialised in 78 ms"
}
2024-04-23T18:15:53+00:00 [INFO] downloading language server "tailwindcss-language-server"
2024-04-23T18:15:53+00:00 [INFO] Node runtime install_if_needed
2024-04-23T18:15:53+00:00 [INFO] Language server with id 2 sent unhandled notification indexingStarted:
[
  null
]
2024-04-23T18:15:53+00:00 [INFO] starting language server. binary path: "/Users/xcrap/Library/Application Support/Zed/node/node-v18.15.0-darwin-arm64/bin/node", working directory: "/Users/xcrap/htdocs/hotelgaivota", args: ["/Users/xcrap/Library/Application Support/Zed/languages/tailwindcss-language-server/node_modules/.bin/tailwindcss-language-server", "--stdio"]
2024-04-23T18:15:53+00:00 [INFO] Language server with id 2 sent unhandled notification indexingEnded:
[
  null
]
2024-04-23T18:15:54+00:00 [INFO] Language server with id 1 sent unhandled notification @/tailwindCSS/projectInitialized:
[
  null
]


### Environment

Zed: v0.131.7 (Zed)
OS: macOS 14.4.1
Memory: 16 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

_No response_

--- a/crates/project/src/project.rs
+++ b/crates/project/src/project.rs
@@ -4676,12 +4676,21 @@ impl Project {
 
         let mut project_transaction = ProjectTransaction::default();
         for (buffer, buffer_abs_path) in &buffers_with_paths {
-            let adapters_and_servers: Vec<_> = project.update(&mut cx, |project, cx| {
-                project
-                    .language_servers_for_buffer(&buffer.read(cx), cx)
-                    .map(|(adapter, lsp)| (adapter.clone(), lsp.clone()))
-                    .collect()
-            })?;
+            let (primary_adapter_and_server, adapters_and_servers) =
+                project.update(&mut cx, |project, cx| {
+                    let buffer = buffer.read(cx);
+
+                    let adapters_and_servers = project
+                        .language_servers_for_buffer(buffer, cx)
+                        .map(|(adapter, lsp)| (adapter.clone(), lsp.clone()))
+                        .collect::<Vec<_>>();
+
+                    let primary_adapter = project
+                        .primary_language_server_for_buffer(buffer, cx)
+                        .map(|(adapter, lsp)| (adapter.clone(), lsp.clone()));
+
+                    (primary_adapter, adapters_and_servers)
+                })?;
 
             let settings = buffer.update(&mut cx, |buffer, cx| {
                 language_settings(buffer.language(), buffer.file(), cx).clone()
@@ -4734,10 +4743,8 @@ impl Project {
             // Apply language-specific formatting using either the primary language server
             // or external command.
             // Except for code actions, which are applied with all connected language servers.
-            let primary_language_server = adapters_and_servers
-                .first()
-                .cloned()
-                .map(|(_, lsp)| lsp.clone());
+            let primary_language_server =
+                primary_adapter_and_server.map(|(_adapter, server)| server.clone());
             let server_and_buffer = primary_language_server
                 .as_ref()
                 .zip(buffer_abs_path.as_ref());

